straxen/plugins/event_processing.py

```diff
@@ -322,38 +330,63 @@ class EventPositions(strax.Plugin):
          'Correction added to r_naive for field distortion (cm)'),
         ('theta', np.float32,
          'Interaction angular position (radians)')
-    ] + strax.time_fields
+            ] + strax.time_fields
 
     def setup(self):
-        self.map = InterpolatingMap(
-            get_resource(self.config['fdc_map'], fmt='binary'))
+
+        is_CMT = isinstance(self.config['fdc_map'], tuple)
+
+        if is_CMT:
+            
+            map_algo = list(self.config['fdc_map'])
+            name_map = list(map_algo[1])
+            name_map[0] = '_'.join([name_map[0], self.config['default_reconstruction_algorithm']])
+            map_algo[1] = tuple(name_map)
+            map_algo = tuple(map_algo)
+            
+            self.map = InterpolatingMap(
+                get_resource(get_config_from_cmt(self.run_id, map_algo), fmt='binary'))
+
+        elif isinstance(self.config['fdc_map'], str):
+            self.map = InterpolatingMap(
+                get_resource(self.config['fdc_map'], fmt='binary'))
+
+        else:
+            raise NotImplementedError('FDC map format not understood.')
+
+        self.map.scale_coordinates([1., 1., -self.config['electron_drift_velocity']])
```

Comment:
Is scaling always needed? Also for the 1T maps? Otherwise you might add it under the `if is_CMT` block

Reply:
Indeed it is not. When I checked if XENON1T data looked fine I only looked at the (x,y) distribution, but I guess the (r², z) would have shown that this is actually something wrong.